### PR TITLE
Adding ProgressCharacter to Write-Progress and ProgressRecord.

### PR DESCRIPTION
**Pull request by adamdriscoll**
Adds "ProgressCharacter" to the ProgressRecord class and a "ProgressCharacter" parameter to the Write-Progress cmdlet. 

This addresses: #2488

image

### CODE REVIEW

**Comment by adamdriscoll** [comment]
This PR does not include tests. I'm actually not quite sure how to test this appropriately and would be happy to add some tests with some direction.

**Comment by adamdriscoll** [comment]
The only issue with this PR is that it really doesn't fix the fact that binary cmdlets could write progress with a different character. If that's something required by the original issue, we probably just want to close this PR and figure out a different solution.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WriteProgressCmdlet.cs [hunk 1]
```diff
@@ -142,6 +142,16 @@ public SwitchParameter Completed
...
 
+        /// <summary>
+        ///
+        /// Character used to display progress.
+        ///
+        /// </summary>
+        /// <value></value>
+
+        [Parameter]
+        public char ProgressCharacter { get; set; } = 'o';
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WriteProgressCmdlet.cs]: If we already have the default in ProgressRecord.cs (`private char _progressCharacter = 'o'; `) so it should be removed here.

**Comment by iSazonov** [comment]
@adamdriscoll We could extend RawUI in `TestHost` to write tests for `WriteProgress`. See [Read-Host.Tests.ps1](https://github.com/PowerShell/PowerShell/blob/master/test/powershell/Modules/Microsoft.PowerShell.Utility/Read-Host.Tests.ps1) And it is likely that different tests are needed for different platforms.
Maybe @JamesWTruher comment this.

About binary cmdlets. You add `ProgressCharacter `  to public class `ProgressRecord` so binary cmdlets can use free the property.

**Comment by powercode** [comment]
But does this really belong to the progress record? Isn't that something that should be on the host responsible for rendering the progress?

**Comment by adamdriscoll** [comment]
@powercode I kind of thought that was well after implementing this. The ProgressNode (extends from ProgressRecord) that actually does the rendering of the progress has access to the PSHostRawUserInterface for the host. We could add a property onto there to adjust the character as well.

**Comment by lzybkr** [comment]
You will need to test this change with remoting between versions of PowerShell - the type is known to remoting and older versions of PowerShell may not handle the new property.

**Comment by adamdriscoll** [comment]
@lzybkr Noted. What's your thought on how this is implemented? Think it should be more of a host thing and not on the ProgressRecord itself?

**Comment by lzybkr** [comment]
It does seem like a host thing - e.g. a graphical host wouldn't use a character.

**Comment by adamdriscoll** [comment]
I'm gonna close this and look at implementing it differently.

**Pull request closed without merging**